libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -0,0 +1,184 @@
+        return data
+
+    @RestTestLib.authenticate_and_login
+    def get_system_topology(self, auth_header=None, query_param=None):
+        """
+        Get system topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Get system topology request....")
+        if auth_header is None:
+            headers = self.headers
+        else:
+            headers = auth_header
+        endpoint = self.config["get_system_topology"]
+        if query_param is not None:
+            endpoint = endpoint + '/' + query_param
+        response = self.restapi.rest_call("get", endpoint=endpoint,
+                                          headers=headers)
+        self.log.info("Get system topology request successfully sent...")
+        return response
+
+    # Function not ready
+    def verify_system_topology(self, expected_response=HTTPStatus.OK):
+        """
+        Verify Get system topology
+        """
+        resp = self.get_system_topology()
+        result = True
+        if resp.status_code == expected_response:
+            result = True
+            get_response = resp.json()
+            self.log.info("Verify id fields have unique values")
+            self.log.info("Verfying cvg and devices in resp and solution yaml")
+            # Verify deployment version and deployment time
+        else:
+            self.log.error("Status code check failed.")
+            result = False
+        return result, get_response
+
+    @RestTestLib.authenticate_and_login
+    def get_cluster_topology(self, auth_header=None, cluster_id=None):
+        """
+        Get cluster topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Get cluster topology request....")
+        query_param = 'cluster'
+        if cluster_id is not None:
+            query_param = query_param + '/' + cluster_id
+        response = self.get_system_topology(auth_header, query_param)
+        return response
+
+    @RestTestLib.authenticate_and_login
+    def get_storage_topology(self, auth_header=None, storage_set_id=None):
+        """
+        Get cluster id topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Get cluster topology request....")
+        query_param = 'storage_set'
+        if storage_set_id is not None:
+            query_param = query_param + '/' + storage_set_id
+        response = self.get_cluster_topology(auth_header, query_param)
+        return response
+
+    @RestTestLib.authenticate_and_login
+    def get_node_topology(self, auth_header=None, node_id=None, cluster_id=None):
+        """
+        Get node id topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Get cluster topology request....")
+        query_param = cluster_id + '/' + 'node'
```

Comment:
pass cluster_id to get_cluster_topology directly

Reply:
done